v22.1/security-reference/scram-authentication.md

```diff
@@ -0,0 +1,323 @@
+
+Note the importance of rotating passwords twice, both in steps (1) and again in (3), with different passwords.  Step (1) ensures that clients at step (2) can continue to connect after non-SCRAM logins are blocked. However, between step (1) and step (2) it is still possible for CockroachDB to receive plaintext passwords from clients, and so the passwords used at step (1) are not fully protected yet. After step (2), CockroachDB will not learn plaintext passwords from clients any more; then after step 3 it will not know plaintext passwords at all.
+
+### Configuring your cluster settings for strict credential isolation
+
+To enable strict credential isolation from CockroachDB, additional cluster settings are required, above those detailed for simply enabling SCRAM authentication, as detailed [above](#enabling-scram-sha-256-authentication-for-new-users-roles).
+
+Required cluster settings:
+
+
+#### `server.user_login.password_encryption`
+
+Enable SCRAM for new passwords.
+
+{% include_cached copy-clipboard.html %}
+```sql
+SET CLUSTER SETTING server.user_login.password_encryption = 'scram-sha-256';
+```
+
+#### `server.user_login.store_client_pre_hashed_passwords.enabled`
+
+Enable the CockroachDB to accept pre-computed SCRAM-SCHA-256 salted password hashes, rather than accepting a plaintext password and computing the hash itself.
+
+{% include_cached copy-clipboard.html %}
+```sql
+SET CLUSTER SETTING setting server.user_login.store_client_pre_hashed_passwords.enabled = true;
+```
+
+#### `server.host_based_authentication.configuration`
+
+Allow *only* `scram-sha-256` (not the weaker `password`) in your [Authentication Configuration (HBA)](authentication.html#authentication-configuration).
+
+{{site.data.alerts.callout_danger}}
+After this change, users will not be able to sign in with plaintext passwords. All critical SQL users must be migrated to SCRAM-SHA-256 authentication before making this change.
+{{site.data.alerts.end}}
+
+{% include_cached copy-clipboard.html %}
+```sql
+SET CLUSTER SETTING server.host_based_authentication.configuration TO '
+# TYPE    DATABASE  USER          ADDRESS     METHOD
+# any other higher-priority rules ...
+  host    all       human         all         scram-sha-256
+  host    all       app_user      all         cert
+  host    all       cyborg        all         cert-scram-sha-256
+# any other lower-priority rules ...
+```
+
+### Secure handling of plaintext credentials
+
+Using SCRAM authentication, a CockroachdDB cluster need not store or handle the plaintext passwords used by clients to authenticate. When managing SQL users and roles, only salt and hash(salt, password) need be stored.
+
+However, to achieve full separation of concerns, the plaintext credentials must be isolated to the extent possible: they must be exposed as little as possible in transit, storage and during computation.
```

Comment:
The colon here makes it a little difficult to read with "to the extent possible" preceding. Do you think this could be phrased without that? 

"However, to achieve full separation of concerns, the plaintext credentials must be isolated so that they are exposed as little as possible in transit, storage, and during computation."